File: src/leetcode/2026/05/Solution883.java
public class Solution883 {
    public int projectionArea(int[][] grid) {
        int n = grid.length;
        int xy = 0, xz = 0, yz = 0;

        for (int i = 0; i < n; i++) {
            int maxRow = 0, maxCol = 0;
            for (int j = 0; j < n; j++) {
                if (grid[i][j] > 0)
                    xy++;
                maxRow = Math.max(maxRow, grid[i][j]);
                maxCol = Math.max(maxCol, grid[j][i]);
            }
            xz += maxRow;
            yz += maxCol;
        }

        return xy + xz + yz;
    }

    public static void main(String[] args) {
        Solution883 solution = new Solution883();

        int[][] grid1 = { { 1, 2 }, { 3, 4 } };
        System.out.println(solution.projectionArea(grid1)); // Output: 17

        int[][] grid2 = { { 2 } };
        System.out.println(solution.projectionArea(grid2)); // Output: 5

        int[][] grid3 = { { 1, 0 }, { 0, 2 } };
        System.out.println(solution.projectionArea(grid3)); // Output: 8
    }
}

File: src/leetcode/2026/05/Solution883.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution883 {
public:
    int projectionArea(vector<vector<int>>& grid) {
        int n = grid.size();
        int xy = 0, yz = 0, zx = 0;

        for (int i = 0; i < n; ++i) {
            int maxRow = 0, maxCol = 0;
            for (int j = 0; j < n; ++j) {
                if (grid[i][j] > 0) {
                    ++xy;
                }
                maxRow = max(maxRow, grid[i][j]);
                maxCol = max(maxCol, grid[j][i]);
            }
            yz += maxRow;
            zx += maxCol;
        }

        return xy + yz + zx;
    }
};

int main() {
    Solution883 solution;

    vector<vector<int>> grid = {{1, 2}, {3, 4}};
    cout << solution.projectionArea(grid) << endl; // Output: 17

    grid = {{2}};
    cout << solution.projectionArea(grid) << endl; // Output: 5

    grid = {{1, 0}, {0, 2}};
    cout << solution.projectionArea(grid) << endl; // Output: 8
    return 0;
}